src/main/java/com/exactpro/epfast/template/assertion/InstrEqualityChecker.java

```diff
@@ -0,0 +1,160 @@
+                        if (!asciiStringEquality((AsciiStringField) actualItem, (AsciiStringField) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "UnicodeStringField":
+                        if (!unicodeStringEquality((UnicodeStringField) actualItem, (UnicodeStringField) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "ByteVectorField":
+                        if (!byteVectorEquality((ByteVectorField) actualItem, (ByteVectorField) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "CompoundDecimalField":
+                        if (!compoundDecEquality((CompoundDecimalField) actualItem, (CompoundDecimalField) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "SimpleDecimalField":
+                        if (!simpleDecEquality((SimpleDecimalField) actualItem, (SimpleDecimalField) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "Group":
+                        if (!groupEquality((Group) actualItem, (Group) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    case "Sequence":
+                        if (!sequenceEquality((Sequence) actualItem, (Sequence) instrItem)) {
+                            return false;
+                        }
+                        break;
+                    default:
+                        return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private static boolean templateRefEquality(TemplateRef actual, TemplateRef instr) {
+        return areSameReferences(actual.getTemplateRef(), instr.getTemplateRef());
+    }
+
+    private static boolean int32Equality(Int32Field actual, Int32Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean int64Equality(Int64Field actual, Int64Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean uInt32Equality(UInt32Field actual, UInt32Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean uInt64Equality(UInt64Field actual, UInt64Field instr) {
+        return fieldInstrEquality(actual, instr) && areSameOperators(actual.getOperator(), instr.getOperator());
+    }
+
+    private static boolean asciiStringEquality(AsciiStringField actual, AsciiStringField instr) {
```

Comment:
You can use one name for all methods, i.e. `equals`. Java will distinguish them by parameter types.